### PR TITLE
Bindings factory package

### PR DESCRIPTION
**Pull request by jacoscaz**
This PR represents step 1. of the following plan:

1. centralize the implementation of the factory function for `Bindings` objects as a separate package
2. make all the actors in Comunica and `@comunica/sparqlee` depend on the centralized implementation, thus decoupling them from the details of how the factory function is implemented (i.e. via Immutable.js)

This will support later efforts focused on reducing the impact on bundle size of the factory implementation, which will likely consist of work done on Immutable's support for tree-shaking.

~~This PR is currently a draft, aiming to gather some feedback before refactoring more of Comunica's packages.~~

### CODE REVIEW

File: packages/bindings-factory/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Comunica Bindings Factory
+
+[TBD]
```
Comment (rubensworks) [inline packages/bindings-factory/README.md:3]:
```suggestion
This package provides a factory for `Bindings` objects, which allow variables to be mapped to RDF terms.
```

File: packages/bindings-factory/lib/BindingsFactory.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { Map } from 'immutable';
+
+export class BindingsFactory {
+  /**
+   * A convenience constructor for bindings based on a given hash.
+   * @param {{[p: string]: RDF.Term}} hash A hash that maps variable names to terms.
+   * @return {Bindings} The immutable bindings from the hash.
+   * @constructor
+   */
+  // eslint-disable-next-line no-redeclare
```
Comment (rubensworks) [inline packages/bindings-factory/lib/BindingsFactory.ts:12]: Is this eslint-disable still needed?

**Review by rubensworks — CHANGES_REQUESTED** [review]
Thanks for this! :-)

Is it possible that you accidentally rebased on `master`? Because it looks like one of your commits includes changes from `master` that have not been merged yet into `next/major`. Could you undo this, or restore the original commits from `master`? Otherwise we'll have problems when merging `master` and `next/major`.

**Comment by jacoscaz** [comment]
> Is it possible that you accidentally rebased on master

I'm not sure what happened exactly. I did rebase from `master` to `next/major` last week and everything looked ok but you are right, there are changes from the last couple of commits on `master` that should not be in this branch. I'll try to revert them manually...

**Pull request merged**